apm: Document sampling.tail.discard_on_write_failure config (#1453)
elastic/docs-content: solutions/observability/apm/tail-based-sampling.md

| APM Server binary | `sampling.tail.policies` |
| Fleet-managed | `Policies` |

### Discard On Write Failure [sampling-tail-discard-on-write-failure-ref]

> Defines the indexing behavior when trace events fail to be written to storage (e.g. when the storage limit is reached). When set to `false`, traces will be indexed, significantly increasing the indexing load. When set to `true`, traces will be discarded, there will be data loss potentially resulting in broken traces.

Reviewer: ```suggestion
Defines the indexing behavior when trace events fail to be written to storage (e.g. when the storage limit is reached). When set to `false`, traces will be indexed regardless of the configured sample rate in policies, significantly increasing the indexing load. When set to `true`, traces will be discarded, there will be data loss potentially resulting in broken traces.
```
nit on description. Trying to make the implication clear. Feel free to change.

Author: Makes I added a note to specify we bypass sampling

Default: `false`. (bool)

|                              |                                          |
|------------------------------|------------------------------------------|

> | APM Server binary            | `sampling.tail.discard_on_write_failure` |

Reviewer: ```suggestion
| APM Server binary            | `apm-server.sampling.tail.discard_on_write_failure` |
```
Align with https://github.com/elastic/docs-content/pull/2038

> | Fleet-managed (version 9.1+) | `Discard On Write Failure`               |

Reviewer: ```suggestion
| Fleet-managed {applies_to}`stack: ga 9.1` | `Discard On Write Failure`               |
```
corresponds to review comment in https://github.com/elastic/docs-content/pull/1269#discussion_r2185883848
